OTVision/detect/rtsp_input_source.py

```diff
@@ -241,3 +249,48 @@ def create_output(self) -> str:
 
 def convert_frame_to_rgb(frame: ndarray) -> ndarray:
     return cvtColor(frame, COLOR_BGR2RGB)
+
+
+def is_connection_available(rtsp_url: str) -> bool:
+    """
+    Check if RTSP connection is available by sending a DESCRIBE request.
+
+    Args:
+        rtsp_url: The RTSP URL to check
+
+    Returns:
+        bool: True if stream is available, False otherwise
+    """
+    try:
+        parsed = urlparse(rtsp_url)
+        host = parsed.hostname or "localhost"
```

Comment:
Why do we check localhost and port 554 as default?
I think we should fail if hostname or port are not given